### ISSUE TITLE
Error: operation timed out after 4628 ms, 5 tries with error: Status code 403 received from HTTPS request

### ISSUE THREAD

**Issue by TimCody** [cyraxx/pogobuf#59]
- I do not have 2FA
 - I got an email saying new sign in from google.  
 - I know my user name / password / lat /l ong are all correct.  
 - I turned off my firewall
 - Why do I get the timeout error?  

I used the code directly from the inventory.js example.

**Comment by WillCMcC** [comment]
No Pokemon Go APIs are working right now while. Check on the other issues for more info / updates.

**Comment by TimCody** [comment]
Hey,

This one seems to work fine.  

https://pokeadvisor.com/

That's why I thought it was something I'm doing.

**Comment by MRokas** [comment]
> Status code 403 received from HTTPS request

403 = Forbidden

This mean your IP has been blocked.

**Comment by Infient** [comment]
The whole API is not broken. Every request sent by the real Pokémon GO application (on mobile) has an extra parameter – Unknown6 – which is a signature for the servers to check, if the request is valid (sent by real PoGO application). This validation process was not "turned on" until recently, but only for getting map objects.

So, you should be able to do everything just fine, except getting map objects (Pokémons, pokéstops and gyms). The request validation could be turned on to all request at any time. There are huge amount of developers (https://github.com/cyraxx/pogobuf/issues/57#issuecomment-237579182) trying to find how this signature parameter works and how it could be replicated to simulate real application to make APIs work again.

403 status occurs when you are doing something you should not do. Even if you do not have 2FA enabled, Google could block your logins due to region limiting. This is a security precaution that blocks logins from other countries you typically log in – unless you verify the login with other authentication methods (phone, 2FA, manual, etc.). To avoid region limiting you should run the API in the same country you normally use your account. (i.e. You use your account regularly in London, but you run your script in a server located in Sydney -> region limiting)

If you are using servers from one of [the blocked ISPs](https://github.com/cyraxx/pogobuf/wiki/Blocked-server-ISPs), you should change your provider or use proxies due to them being blocked by Niantic to stop APIs working with large VPS providers.

This library tries to retry the request 5 times (default) if the request was not successful and then it throws a timeout error.

For future, you should use [the status code table](https://github.com/cyraxx/pogobuf/wiki/Status-codes) to determine issues regarding status codes.

tl;dr: [the status code table](https://github.com/cyraxx/pogobuf/wiki/Status-codes); your server is blocked or you are logging in from a different country you usually log in

**Comment by cyraxx** [comment]
Thanks for the detailed writeup @joelfi! I'd like to add that since you're getting the error 403 from the retry code ("5 tries with error") it has to be the actual API request that's failing and not the login since the retry logic doesn't exist in the login classes.

Because retrying in this case doesn't make any sense, we should also patch the client so that a `StopError` is thrown instead of `Error` when the HTTP status code from the API is 403, or maybe even everything in the 400-499 range.
